neutrino.go

```diff
@@ -1081,6 +1081,33 @@ func (s *ChainService) BanPeer(addr string, reason banman.Reason) error {
 	return s.banStore.BanIPNet(ipNet, reason, BanDuration)
 }
 
+// UnbanPeer connects and unbans a previously banned peer.
+func (s *ChainService) UnbanPeer(addr string) error {
+	log.Warnf("UnBanning peer %v", addr)
+
+	defer func() error {
+		if sp := s.PeerByAddr(addr); sp != nil {
+			tcpAddr, err := s.addrStringToNetAddr(addr)
+			if err != nil {
+				return err
+			}
+			s.connManager.Connect(&connmgr.ConnReq{
+				Addr:      tcpAddr,
+				Permanent: true,
```

Comment:
Perhaps this should be optional? So a bool that controls if we'll try to permanently connecvt to them after banning.

Reply:
I don't think this defer is necessary?

Reply:
Are you saying we should skip the defer function and just connect the peer that has just been unbanned?